classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -488,4 +495,74 @@ protected function setFallbackVersion(): ?string
         }
         return null;
     }
+
+    /**
+     * Check the contexts' contact details before upgrade
+     *
+     * @return void
+     * @throws Exception
+     */
+    protected function checkContactSetting(): void
+    {
+        $primaryLocale = DB::table('site')
+            ->select(['primary_locale'])
+            ->first()
+            ->primary_locale;
+
+        // Get all the enabled context with context name/title
+        $contexts = DB::table("{$this->getContextTable()} AS context")
+            ->select([
+                "context.{$this->getContextKeyField()}", 
+                "context.enabled", 
+                "context_setting.locale",
+                "context_setting.setting_name",
+                "context_setting.setting_value"
+            ])
+            ->join("{$this->getContextSettingsTable()} AS context_setting", fn($join) =>
+                $join
+                    ->on("context.{$this->getContextKeyField()}", "=", "context_setting.{$this->getContextKeyField()}")
+                    ->where("context_setting.locale", "=", $primaryLocale)
+                    ->where("context_setting.setting_name", "=", "name")
+            )
+            ->where("context.enabled", 1)
+            ->get();
+        
+        // Get all the context IDs for which contact name/email is set
+        $contextIdsWithValidContact = DB::table("{$this->getContextTable()} AS context")
+            ->select(["context.{$this->getContextKeyField()}"])
+            ->join("{$this->getContextSettingsTable()} AS contact_name_setting", fn($join) => 
+                $join
+                    ->on("contact_name_setting.{$this->getContextKeyField()}", "=", "context.{$this->getContextKeyField()}")
+                    ->where("contact_name_setting.setting_name", "=", "contactName")
+            )
+            ->leftJoin("{$this->getContextSettingsTable()} AS contact_email_setting", fn($join) => 
+                $join
+                    ->on("contact_email_setting.{$this->getContextKeyField()}", "=", "context.{$this->getContextKeyField()}")
+                    ->where("contact_email_setting.setting_name", "=", "contactEmail")
+            )
+            ->whereNotNull("contact_email_setting.{$this->getContextKeyField()}")
+            ->whereIn("context.{$this->getContextKeyField()}", $contexts->pluck("{$this->getContextKeyField()}")->toArray())
```

Comment:
Is this to ensure that only enabled journals are retrieved?
What about using direct SQL transaction to find `journal_id`'s where contactName and contactEmail don't exist? Considering the previous example:
```sql
SELECT `journal_id` FROM `journals` WHERE `journal_id` NOT IN (SELECT `journal_id` FROM `journal_settings` WHERE `setting_name` IN ('contactEmail', 'contactName')) AND `enabled` = 1
....
```
For the message, you then can retrieve the journal name by its ID

Reply:
Yeah, we only need to target the enable ones , right ? Or should we test all existing journals? Extra overhead can be associated in both case as if system have one or multiple disable contexts, then

1. If no plan to activate those in future, make no sense for them to be forced to have those information added during the upgrade 
2. If system admin has the plan to active those in future and not have these info added during upgrade but enable the contexts later after upgrade and then not providing those contact name/email, email and submission will not work properly and they may have a hard to track the issue . 

@Vitaliy-1   What is your suggestion in such case ?

Reply:
Thinking about this further, even if the journal is disabled from appearing publicly, it still can be used to send emails and some of them still rely on a context contact email. At least the system allows that. I think it makes sense to include them in the check.

Reply:
got it, removed the `enable` constrain.